mysql-test/suite/galera/t/galera_concurrent_ctas.test

```diff
@@ -1,56 +1,53 @@
 --source include/galera_cluster.inc
 --source include/big_test.inc
+--source include/have_debug_sync.inc
+
+#
+# To have real concurrent CTAS this test uses DEBUG_SYNC
+#
+
+--connect node_1a, 127.0.0.1, root, , test, $NODE_MYPORT_1
+
+--connection node_1
+#
+# Send CTAS it will block before open_tables call
+#
+SET DEBUG_SYNC = 'wsrep_create_table_as_select WAIT_FOR continue';
+--send CREATE table t1 as SELECT SLEEP(0.1);
+
+--connection node_1a
+#
+# Wait for CTAS to block
+#
+--echo # Wait until CTAS is on debug sync point
+--let $wait_condition = SELECT COUNT(*) = 1 FROM INFORMATION_SCHEMA.PROCESSLIST WHERE STATE = 'debug sync point: wsrep_create_table_as_select'
+--source include/wait_condition.inc
+
+--connection node_2
+#
+# Send concurrent CTAS
+#
+--send CREATE table t1 as SELECT SLEEP(0.2);
+
+--connection node_1a
+--echo # Wait until CTAS has executed
+SET DEBUG_SYNC= 'now SIGNAL continue';
```

Comment:
the CREATE TABLE send from node_2 is not waited here. There is no guarantee if this WS has arrived in node_1 or how far it has processed before the first CREATE TABLE execution is signaled to continue

Reply:
This is true and I will fix it.